metadata-ingestion/tests/unit/test_kafka_source.py

```diff
@@ -26,407 +25,463 @@
 )
 
 
```

Comment:
Kafka source unit tests did not execute earlier as part of unit tests, since KafkaSourceTest did not inherit from unittest.TestCase.
Also parameterize does not work with unit test class (https://pytest.org/en/7.2.x/how-to/unittest.html#pytest-features-in-unittest-testcase-subclasses), so moving these out of the Test class.